src/library_syscall.js

```diff
@@ -924,138 +903,134 @@ var SyscallsLibrary = {
   },
-  __syscall191: function(which, varargs) { // ugetrlimit
+  __syscall191: function(which, resource, rlim) { // ugetrlimit
 #if SYSCALL_DEBUG
     err('warning: untested syscall');
 #endif
-    var resource = SYSCALLS.get(), rlim = SYSCALLS.get();
     {{{ makeSetValue('rlim', C_STRUCTS.rlimit.rlim_cur, '-1', 'i32') }}};  // RLIM_INFINITY
     {{{ makeSetValue('rlim', C_STRUCTS.rlimit.rlim_cur + 4, '-1', 'i32') }}};  // RLIM_INFINITY
     {{{ makeSetValue('rlim', C_STRUCTS.rlimit.rlim_max, '-1', 'i32') }}};  // RLIM_INFINITY
     {{{ makeSetValue('rlim', C_STRUCTS.rlimit.rlim_max + 4, '-1', 'i32') }}};  // RLIM_INFINITY
     return 0; // just report no limits
   },
   __syscall192__deps: ['_emscripten_syscall_mmap2'],
-  __syscall192: function(which, varargs) { // mmap2
-    var addr = SYSCALLS.get(), len = SYSCALLS.get(), prot = SYSCALLS.get(), flags = SYSCALLS.get(), fd = SYSCALLS.get(), off = SYSCALLS.get()
+  __syscall192: function(which, addr, len, prot, flags, fd, off) { // mmap2
     return __emscripten_syscall_mmap2(addr, len, prot, flags, fd, off);
   },
-  __syscall193: function(which, varargs) { // truncate64
-    var path = SYSCALLS.getStr(), zero = SYSCALLS.getZero(), length = SYSCALLS.get64();
+  __syscall193: function(which, path, zero, low, high) { // truncate64
+    path = SYSCALLS.getStr(path);
+    var length = SYSCALLS.get64(low, high);
     FS.truncate(path, length);
     return 0;
   },
-  __syscall194: function(which, varargs) { // ftruncate64
-    var fd = SYSCALLS.get(), zero = SYSCALLS.getZero(), length = SYSCALLS.get64();
+  __syscall194: function(which, fd, zero, low, high) { // ftruncate64
+    var length = SYSCALLS.get64(low, high);
     FS.ftruncate(fd, length);
     return 0;
   },
-  __syscall195: function(which, varargs) { // SYS_stat64
-    var path = SYSCALLS.getStr(), buf = SYSCALLS.get();
+  __syscall195: function(which, path, buf) { // SYS_stat64
+    var path = SYSCALLS.getStr(path);
     return SYSCALLS.doStat(FS.stat, path, buf);
   },
-  __syscall196: function(which, varargs) { // SYS_lstat64
-    var path = SYSCALLS.getStr(), buf = SYSCALLS.get();
+  __syscall196: function(which, path, buf) { // SYS_lstat64
+    var path = SYSCALLS.getStr(path);
     return SYSCALLS.doStat(FS.lstat, path, buf);
   },
-  __syscall197: function(which, varargs) { // SYS_fstat64
-    var stream = SYSCALLS.getStreamFromFD(), buf = SYSCALLS.get();
+  __syscall197: function(which, fd, buf) { // SYS_fstat64
+    var stream = SYSCALLS.getStreamFromFD(fd);
     return SYSCALLS.doStat(FS.stat, stream.path, buf);
   },
-  __syscall198: function(which, varargs) { // lchown32
-    var path = SYSCALLS.getStr(), owner = SYSCALLS.get(), group = SYSCALLS.get();
+  __syscall198: function(which, path, owner, group) { // lchown32
+    path = SYSCALLS.getStr(path);
     FS.chown(path, owner, group); // XXX we ignore the 'l' aspect, and do the same as chown
     return 0;
   },
-  __syscall199__sig: 'iii',
+  __syscall199__sig: 'i',
```

Comment:
this should be `ii` - one return value, one param. likewise the ones below in this section.

Reply:
Is there a benefit to these `__sig`s there.  I seems they are used only the syscalls that we don't actually implement.  If its an optimization it seems misguided since we should never actually hit these functions in a normal running program.

Reply:
The main use is these aliases: we use the sig to construct a proper function to call the other one. Without the proper args the VM might be slower, and closure might not merge the two.

Note that this goes the other way from what I said earlier about not adding unnecessary params in syscalls. I think since in the other case we're calling from wasm into JS anyhow, the extra overhead is not significant. However, that might have changed since last I checked, so it might be safer to still include those for now.